src/backend/distributed/utils/query_utils.c

```diff
@@ -132,6 +132,31 @@ ExtractRangeTableRelationWalker(Node *node, List **rangeTableRelationList)
 }
 
 
+/*
+ * SplitIntoQueries returns a list of queries by splitting
+ * the given concatenated query string with delimiter ';'
+ */
+List *
+SplitIntoQueries(char *concatenatedQueryString)
+{
+	List *queries = NIL;
+	rsize_t len = (rsize_t) strlen(concatenatedQueryString);
+	char *delimiter = ";";
+	char *remaining = concatenatedQueryString;
+	char *query = strtok_s(concatenatedQueryString, &len, delimiter, &remaining);
```

Comment:
this will not be correct if a query has ';' in it other than as a separator.

Reply:
Is it possible to use ";" (e.g.) for a table name ? I ask this to be sure that if it is ever possible to see ";" other than as a separator ?

Reply:
maybe not table names but schemas can have that:

```
talha=# create schema "talha;;"; 
CREATE SCHEMA                    
```

Reply:
thanks for this clarification

Reply:
I removed this method and used `task->queryStringList`. This is not an issue anymore.

@marcocitus  rebased the PR with `task->queryStringList` change.